openvdb/points/IndexFilter.h

```diff
@@ -142,15 +200,23 @@ class MultiGroupFilter
 
     inline bool initialized() const { return mInitialized; }
 
+    inline index::State state() const
+    {
+        return (mInclude.empty() && mExclude.empty()) ? index::ALL : index::PARTIAL;
+    }
+
+    template <typename LeafT>
+    static index::State state(const LeafT&) { return index::PARTIAL; }
```

Comment:
Why not use the leaf here to optimise further? See above comment

Reply:
Again, happy to accept a contribution here. :)

Reply:
Potentially something that should be done as part of the PR?